addons/l10n_ec/migrations/3.6/post-migrate_update_taxes.py

```diff
@@ -0,0 +1,64 @@
+            (f'{company.id}_tax_withhold_profit_3440', '3440 2.75% Otras Retenciones Aplicables el 2,75%')
+        ]
+        for xml_id, name in taxes_to_update:
+            cr.execute("UPDATE account_tax SET name=%s "
+                       "WHERE id=(SELECT res_id FROM ir_model_data WHERE module = 'l10n_ec' AND name=%s)", (name, xml_id))
+
+def update_ec_codes(cr, env):
+    """
+    Update special fields for Ecuador, the code base and code applied in the taxes
+    """
+    companies = env['res.company'].search([('chart_template_id', '=', env.ref('l10n_ec.l10n_ec_ifrs').id)])
+    for company in companies:
+        taxes_to_update = [
+            (f'{company.id}_tax_withhold_profit_3440', '3340', '3940')
+        ]
+        for xml_id, l10n_ec_code_base, l10n_ec_code_applied in taxes_to_update:
+            cr.execute("UPDATE account_tax SET l10n_ec_code_base=%s, l10n_ec_code_applied=%s "
+                       "WHERE id=(SELECT res_id FROM ir_model_data WHERE module = 'l10n_ec' AND name=%s)",
+                       (l10n_ec_code_base, l10n_ec_code_applied, xml_id))
+
+def inactivate_taxes_replaced(cr, env):
+    """
+    Inactivate the taxes that were replaced by the new ones
+    """
+
+    companies = env['res.company'].search([('chart_template_id', '=', env.ref('l10n_ec.l10n_ec_ifrs').id)])
+    for company in companies:
+        taxes_to_inactivate = [
+            f'{company.id}_tax_withhold_profit_304',  # replaced by tax_withhold_profit_304_10
+            f'{company.id}_tax_withhold_profit_304A',  # replaced by tax_withhold_profit_304A_10
+            f'{company.id}_tax_withhold_profit_304B',  # replaced by tax_withhold_profit_304B_10
+            f'{company.id}_tax_withhold_profit_304E',  # replaced by tax_withhold_profit_304E_10
+            f'{company.id}_tax_withhold_profit_309',  # replaced by tax_withhold_profit_309_2_75
+            f'{company.id}_tax_withhold_profit_314A',  # replaced by tax_withhold_profit_314A_10
+            f'{company.id}_tax_withhold_profit_314B',  # replaced by tax_withhold_profit_314B_10
+            f'{company.id}_tax_withhold_profit_314C',  # replaced by tax_withhold_profit_314C_10
+            f'{company.id}_tax_withhold_profit_314D',  # replaced by tax_withhold_profit_314D_10
+            f'{company.id}_tax_withhold_profit_320',  # replaced by tax_withhold_profit_320_10
+            f'{company.id}_tax_withhold_profit_322',  # replaced by tax_withhold_profit_322_10
+        ]
+        for xml_id in taxes_to_inactivate:
+            cr.execute("UPDATE account_tax SET active=False "
+                       "WHERE id=(SELECT res_id FROM ir_model_data WHERE module = 'l10n_ec' AND name=%s)", (xml_id,))
+def migrate(cr, version):
+    env = api.Environment(cr, SUPERUSER_ID, {})
+
+    update_taxes_from_templates(cr, 'l10n_ec.l10n_ec_ifrs')
+    update_names(cr, env)
+    update_ec_codes(cr, env)
+    inactivate_taxes_replaced(cr, env)
```

Comment:
```suggestion
    update_taxes_from_templates(cr, 'l10n_ec.l10n_ec_ifrs')
```
What does not work if you just do this?

Reply:
In 'update_taxes_from_templates' method when it detects a tax that already exists, it only updates the tax tags and nothing more...So I update the rest in other methods.

Reply:
@rrvc12 Ok, the best way is to keep as is then.  But in saas-16.3 we can just remove all migration scripts then.  (so only needed here)